doUpsert/index.js

```diff
@@ -112,19 +143,26 @@ module.exports = async function(context, jsonItem) {
             });
       })();
     } else { // twin must be upserted
-      context.log.verbose(`upserting twin ${jsonItem.$id}`);
-      // twin
-      await (async () => {
-        context.log.verbose('waiting 20ms');
-        sleep(20);
-        context.log.verbose('calling ADT twin API');
-        await digitalTwin.upsertDigitalTwin(jsonItem.$id, jsonString)
-            .catch((e) => {
-              context.log.error(`twin ${jsonItem.$id} insertion failed: `, e);
-              const err = `failed twin: ${jsonString}`;
-              throw err;
-            });
-      })();
+      try { // if twin already exists, it must be updated instead
+        await digitalTwin.getDigitalTwin(jsonItem.$id);
```

Comment:
Same remark as above, use the returnedValue and do not catch the updateDigitalTwin calls error